packages/fcl/src/exec/query.js

```diff
@@ -43,25 +49,55 @@ import {normalizeArgs} from "./utils/normalize-args"
 export async function query(opts = {}) {
   await preQuery(opts)
 
-  // prettier-ignore
-  return sdk.send([
-    sdk.script(opts.cadence),
-    sdk.args(normalizeArgs(opts.args || [])),
-    opts.limit && typeof opts.limit === "number" && sdk.limit(opts.limit)
-  ]).then(sdk.decode)
+  if (isString(opts?.template)) {
+    opts.template = await retrieve({url: opts?.template})
+  }
+
+  if (opts?.template) {
+    opts.template = normalizeInteractionTemplate(opts?.template)
+  }
+
+  let dependencies = {}
+  if (opts?.template) {
+    dependencies = await deriveDependencies({template: opts.template})
+  }
+
+  const cadence =
+    opts.cadence ||
+    deriveCadenceByNetwork({
+      template: opts.template,
+      network: await sdk.config().get("flow.network"),
+    })
+
+  return config.overload(dependencies, async () =>
+    // prettier-ignore
+    sdk.send([
+      sdk.script(cadence),
+      sdk.args(normalizeArgs(opts.args || [])),
+      opts.limit && typeof opts.limit === "number" && sdk.limit(opts.limit)
+    ]).then(sdk.decode)
+  )
 }
 
 async function preQuery(opts) {
```

Comment:
Should this be prepQuery?

Could this prep be added to utils for use in mutate/query as well?

Reply:
Good call, they're the same checks now. Added!